src/vitabel/timeseries.py

```diff
@@ -2782,21 +2893,53 @@ def toggle_add_text(change):
         add_stack.selected_index = 1
 
         def delete_toggle_handler(change):
-            nonlocal DELETE_ANNOTATIONS
+            nonlocal DELETE_ANNOTATIONS, partial_interval_data, shifting_reference_time
+            partial_interval_data = None             
+            shifting_reference_time = None
+            fig.canvas._figure_label = "."
 
             if change["new"]:  # value of "new" attribute is new button value
                 delete_toggle_button.description = "Mode: Delete Data"
                 delete_toggle_button.button_style = "danger"
                 add_stack.selected_index = 0
                 DELETE_ANNOTATIONS = True
+                active_label = label_dropdown.value
+                if isinstance(active_label, IntervalLabel):
+                    fig.canvas._figure_label = (
+                        "To delete an Interval click close to its start with the right mouse button.")
             else:
                 delete_toggle_button.description = "Mode: Add Data"
                 delete_toggle_button.button_style = "success"
                 add_stack.selected_index = 1
                 DELETE_ANNOTATIONS = False
+                if isinstance(label_dropdown.value, IntervalLabel):
+                    fig.canvas._figure_label = (
+                        "Right-click to set start time, then right-click again to set end time."
+                    )
+        
+        def label_dropdown_change(change):
+            nonlocal DELETE_ANNOTATIONS, partial_interval_data, shifting_reference_time
```

Comment:
`nonlocal` should only be required if you intend to make changes to the variable inside of this scope

```suggestion
            nonlocal partial_interval_data, shifting_reference_time
```